geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java

```diff
@@ -945,17 +945,23 @@ protected void terminateDispatching(boolean checkQueue) {
     }
   }
 
-  private void closeSocket() {
-    if (this._socketClosed.compareAndSet(false, true)) {
+  private boolean closeSocket() {
+    String remoteHostAddress = this._remoteHostAddress;
+    if (this._socketClosed.compareAndSet(false, true) && remoteHostAddress != null) {
```

Comment:
Would it make sense to break the && condition into another if call instead?

if (remoteHostAddress == null) {
  return false?
}

Reply:
I think it's worth comments above each of `closeSocket()` and `closeTransientFields()` explaining that `_socketClosed` is used to provide one-shot protection around the close logic in both methods.